cli/command/image/push.go

```diff
@@ -81,7 +88,7 @@ func RunPush(ctx context.Context, dockerCli command.Cli, opts pushOptions) error
 
 		printNote(dockerCli, `Selecting a single platform will only push one matching image manifest from a multi-platform image index.
 This means that any other components attached to the multi-platform image index (like Buildkit attestations) won't be pushed.
-If you want to only push a single platform image while preserving the attestations, please use 'docker convert\n'
+If you want to push a whole multi-platform image, make sure all image content is present and remove the --platform flag.
```

Comment:
nit: i'd remove `whole` here.   

Also, what does `make sure all image content is present` mean in this scenario?  it's a bit confusing to me

Reply:
You can have a multi-platform image, but only have partial content available.
For example, `busybox` is a multi-platform image, but doing `docker pull busybox` will only fetch the content for your daemon's host platform, assuming that the host is `linux/amd64`:
```
$ docker pull busybox
$ docker pull --platform linux/arm64 busybox
```

Will pull the content for `linux/amd64` and `linux/arm64`, but the "full" `busybox` image index still references all the other supported platforms (`s390x`, `riscv64`, `arm/v5`, ...).

Retagging the busybox as-is and attempting to push it to a different repository in most cases will fail, because the index will reference unknown blobs (of other platforms).

Reply:
bit of an aside, but this `NOTE` color really doesn't look good in my terminal 
<img width="747" alt="image" src="https://github.com/docker/cli/assets/35727626/7b5e4fa9-5f56-4c2f-8f42-b6342f85f02a">

Reply:
I don't really understand what this note is trying to convey. Is it supposed to be a warning? When I read this it feels like I am doing something wrong. But we're not exactly saying that.
Can we clarify somehow?

Reply:
> You can have a multi-platform image, but only have partial content available.
> 
> For example, `busybox` is a multi-platform image, but doing `docker pull busybox` will only fetch the content for your daemon's host platform, assuming that the host is `linux/amd64`:
> 
> ```
> 
> $ docker pull busybox
> 
> $ docker pull --platform linux/arm64 busybox
> 
> ```
> 
> 
> 
> Will pull the content for `linux/amd64` and `linux/arm64`, but the "full" `busybox` image index still references all the other supported platforms (`s390x`, `riscv64`, `arm/v5`, ...).
> 
> 
> 
> Retagging the busybox as-is and attempting to push it to a different repository in most cases will fail, because the index will reference unknown blobs (of other platforms).

Thanks for that explanation @vvoland, makes sense. 
I'm not convinced the terminology we're using is the clearest for the users though (when i hear image content i do not think about architectural variants, but the actual contents of the image - files etc.), but i can't think of many suggestions right now. Where else in the docs do we speak of these things? Do we use "content" with the same meaning in those places?